### PR TITLE
XL Add buildfarm-http-proxy

### PR DESCRIPTION
**Pull request by werkt**
Proxy remote ex requests through to an http endpoint for CAS/AC requests

### CODE REVIEW

File: src/main/java/build/buildfarm/common/UrlPath.java [hunk 1]
```diff
@@ -126,6 +135,17 @@ public static Digest parseUploadBlobDigest(String resourceName, DigestUtil diges
     return digestUtil.build(hash, size);
   }
 
+  public static Digest parseUploadBlobDigest(String resourceName)
```
[the remark this answers is not in the document]
Comment (werkt) [reply src/main/java/build/buildfarm/common/UrlPath.java]: I baked my validation of digest into digestUtil.build, which we've had a discussion about being a restriction (one hash method per instance) that you don't adhere to. You're probably right, this is the right time to abandon that.

**Comment by pcj** [comment]
Any update on this?  @buchgr Have you had a chance to review?

**Comment by buchgr** [comment]
Sorry for the delay. I realistically don't really have cycles to review this any time soon.

**Comment by gdeest-da** [comment]
Is this moving forward ?

If I understand correctly, this PR would enable the use of an HTTP-based cache such as bazel-remote instead of a GRPC-based cache. Is this understanding correct ?

**Comment by werkt** [comment]
I will rebase this shortly, and @buchgr has indicated that some of the things that were adopted from bazel lack some new hotness.  I'll incorporate those prior to merging.

**Pull request merged**